freqtrade/exchange/exchange.py

```diff
@@ -1891,7 +1891,7 @@ def parse_leverage_tier(self, tier) -> Dict:
         return {
             'min': tier['notionalFloor'],
             'max': tier['notionalCap'],
-            'mmr': tier['maintenanceMarginRatio'],
+            'mmr': tier['maintenanceMarginRate'],
```

Comment:
According to the tests against exchanges - this change seems wrong?

Reply:
It's `maintenanceMarginRate` on binance, and it's supposed to be `maintenanceMarginRate`. I posted a [PR](https://github.com/ccxt/ccxt/pull/11983) to CCXT fixing this

This was my mistake, sorry

Reply:
ok then it makes sense
Once it's merged, please bump the ccxt version - if everything goes well, tests should pass then.

Reply:
@samgermain 
looking at the error in the first CI try (`ccxt.base.errors.RateLimitExceeded: okx {"msg":"Requests too frequent.","code":"50011"}`) - maybe okx rate-limits in ccxt are too low (maybe okx changed them recently or the weight for the endpoint is higher than expected)?

We're intentionally not overwriting rateLimit - as ccxt limits are usually OK.

Reply:
The api endpoint for `fetchLeverageTiers` is set up to send a request no more than once every 200 ms which is what is specified in Okx's docs

![](https://i.imgur.com/osIzfiZ.png)

If we have two PRs being tested at once though, then would we be sending requests from the same api on multiple instances of the bot?

Reply:
for sure - but it all runs on the github actions infrastructure - and i don't think it'll be the same exit IP's (that's surely a possibility - but a pretty low one).

It's however a risk we MUST take if we want to test against live exchange endpoints.
we also try to avoid that by killing prior runs on the same branch - so having 2 run in parallel (which actually reach this stage) requires 2 runs to happen from different branches.

Reply:
@xmatthias if you can paste a shortened snippet to reproduce the rate limiting error in py – we will investigate and fix that in ccxt, thx!

Reply:
> requires 2 runs to happen from different branches.

That's what I was referrin to, we got another CCXT member to test out the rate limiter and he also thinks its from running two PRs at the same time also. 

Is it possible to limit this test so that we don't have to test it on python 8, 9 and 10 on windows, mac and linux? I feel like we would run into less problems if we limited it to one operating system. 

We could even set up a russian roulette situation for it to have a 1/3 chance of running, so then the probability of it not being tested would be (2/3)^9 = 2.6%

Reply:
`--longrun` is only running on 3.9 - and only on linux. Can't limit it more than this, really.

the errors are visible in this [ci run](https://github.com/freqtrade/freqtrade/runs/5226543636?check_suite_focus=true). 
It's running on github action cloud workers - which is a shared IP range in Azure datacenters afaik - so in theory it could've been others running tests against okx as well running into this problem.

as far as i can tell, it was our only CI job running at the time (when the job was running, this was the only CI job which had extensive calls against okx).

I'd say we monitor this going forward in feat/short - it was just a "quick suspicion" - i've also not ran into it locally so far.